docs/table/modify_table.rst

```diff
@@ -213,79 +205,34 @@ a time.
 
 .. _table-replace-1_3:
 
-API change in replacing columns
-===============================
+In-place versus replace column update
+=====================================
 
-Astropy version 1.3 introduces an API change in the way that the following
-behaves::
+Consider this code snippet::
 
   >>> t = Table([[1, 2, 3]], names=['a'])
   >>> t['a'] = [10.5, 20.5, 30.5]
 
-Prior to 1.3 this always did an in-place replacement of the data values so that
-the ``t['a']`` column object reference was maintained.  However, since the
-original data type was integer in this case, the replaced values would silently
-be converted to integer by truncation.
+There are a couple of ways this could be handled.  It could update the existing array
+values in-place (truncating to integer), or it could replace the entire column with a new
+column based on the supplied data values.
 
-Starting with astropy 1.3 the operation shown above does a *complete
+The answer for astropy (since version 1.3) is that the operation shown above does a *complete
 replacement* of the column object.  In this case it makes a new column
 object with float values by internally calling
 ``t.replace_column('a', [10.5, 20.5, 30.5])``.  In general this behavior
-is more consistent with Python and Pandas behavior, but there is potential
-for somewhat subtle bugs in code that was written that expects the pre-1.3
-in-place behavior.
-
-**Examples**
-::
-
-  >>> t = Table([[1, 2, 3]], names=['a'])
-  >>> t['a'].description = 'My data column'
-
-  # Sliced column gets replaced
-  >>> t2 = t[:2]  # Make a slice
-
-  # In astropy 1.3 the following emits a warning about replacing a slice.
-  >>> t2['a'] = [10, 20]  # doctest: +SKIP
+is more consistent with Python and Pandas behavior.
 
-  >>> list(t['a'])  # Outputs [10, 20, 3] prior to astropy 1.3.
-  [1, 2, 3]
+**Forcing in-place update**
 
-  # Column reference count changes
-  >>> ta = t['a']  # Make a reference to the original column
-  >>> t['a'] = [10, 20, 30]
-  >>> t['a'] is ta  # Outputs True prior to astropy 1.3
-  False
-
-  # Column attributes change
-  >>> print(t['a'].description)  # Outputs 'My data column' prior to astropy 1.3
-  None
-
-**Replicating pre-1.3 behavior**
```

Comment:
We might wanna keep "Replicating pre-1.3 behavior" section, as I vaguely remember someone recently was looking for this. Surprisingly, users out there might still have old scripts using `astropy<2`. :grimacing:

Reply:
They can still consult an older version of the docs to get those examples?

Reply:
:grimacing:

Reply:
Yeah, I want to cut the cruft where possible.  This was intended as a transition guide for the 99.9% of users that have transitioned.